src/stream/src/common/log_store_impl/kv_log_store/reader.rs

```diff
@@ -355,7 +403,31 @@ impl<S: StateStore> LogReader for KvLogStoreReader<S> {
                 self.rx.truncate(offset);
             }
         }
-        self.truncate_offset = offset;
+        self.truncate_offset = Some(offset);
         Ok(())
     }
+
+    async fn rewind(&mut self) -> LogStoreResult<bool> {
+        self.latest_offset = None;
+        if self.state_store_stream.is_some() {
+            // still consuming persisted state store data
+            let persisted_epoch =
+                self.truncate_offset
+                    .map(|truncate_offset| match truncate_offset {
+                        TruncateOffset::Chunk { epoch, .. } => {
+                            // This is not likely to happen because when consuming state store persisted data,
+                            // we only truncate at barrier
```

Comment:
hmm... Why? I though truncation can happen on any point regardless of whether we are consuming persisted data or not.